### ISSUE TITLE
대여했던 의류를 다시 대여하고자 할때 단축된 절차를 사용하려고 함

### ISSUE THREAD

**Issue by 10001if** [opencloset/opencloset#1340]
#1067 과 연관이 된 이슈입니다.

현재 대여했던 의류를 다시 대여하고자 하는 사람이 본 정보를 입력하여 예약했을 때, 사용자 정보에는 따로 표시가 없고 prepration 화면에 00-00 의류 대여 희망합니다 가 표시되는 걸로 확인 됩니다. 지난 대여 기록으로 의류를 대여하길 희망하는 경우에 해당 정보를 입력받는 것은 다음과 같은 장점을 가지고 있습니다.

    방문 전에 미리 의류를 준비해 놓을 수 있다.
    대리 대여와 유사한 경우입니다. 그 의류가 있다면 동일한 의류를, 해당 의류가 없어도
    신체 사이즈의 변화가 없고 동일하다면 해당 사이즈에 맞는 의류를 미리 준비해 놓을 수 있습니다.

-> 하지만 이를 위해서는, 시간표에 지난 대여의류를 희망하는 사람이라는 안내가 되어야 합니다. (대리대여처럼 표시가 필요)

    구두, 타이처럼 방문시 신체치수 측정 과정 없이 바로 의류준비로 변경되어 대여시간을 단축할 수 있다.
    본 경우도 신체치수 단계가 필요없으므로 대여시간을 단축시켜 오프라인에 더 많은 인원이 대여할 수 있도록 할 수 있습니다.

-> 하지만 이를 위해서는, 현장에서 방문 예약 확인을 하면 #1068 이슈에서처럼 바로 의류준비로 넘어갈 수 있도록 하는 기능이 필요합니다.

**Comment by 10001if** [comment]
원하는 기능은 다음과 같습니다.

 1.   원하는 의류가 있는 경우, 입구에서 예약확인을 진행하면 그 다음 상태가 방문이 아니라 의류준비로 바로 진행될 수 있도록 요청드립니다. 구두, 타이만 하는 경우와 동일한 절차입니다.

  2.  시간표에 대리인과 유사하게, 표시를 하여 시간표만 보고도 (원하는 의류를 입력한 사람이라는 사실을 인지 할 수 있도록) 미리 의류를 준비할 수 있도록 하려고 합니다. 이를 위해 다른 표시를 할 수 있도록 아래와 같이 적용 요청드립니다.

이렇게 바로 찾아야 하는 경우가 1) 방문 예약시에 대여희망을 선택한 경우와 2) 미리 입어본 경우 두 가지가 있습니다. 그런데 옷장지기들이 지난 대여기록을 선택한 대여자인지, 아니면 미리 입어보고 다시 온 경우인지 구분하는 것이 필요합니다.

대리인-대여희망-입어본경우 3가지를 구분하기 위해 (1) 대여희망의 경우 대리인 표시 자리에 마우스 모양을 넣어주시면 될 것 같습니다. 방문 전 대여희망을 마우스로 체크했다고 이해하겠습니다. (2) 그리고 미리 입어본 경우는 RE 표시를 부탁드립니다.

이렇게 두 가지 이슈 적용을 요청드립니다.

**Comment by 10001if** [comment]
2번을 요청드린 이유는 옷을 찾아드리기 위해 방문을 기다리지 않아도 되는 분들인 경우 아침 시간이나 중간에 있는 여유 시간을 활용하여 의류를 미리 준비해 놓기 위함입니다. 이를 위해 위에서 요청드린 시간표 표시 외에 다음을 요청드립니다.

1. 시간표 상에 미리 입어보거나, 대여를 희망하는 사람들의 인원을 표시하고자 합니다. 둘의 합계 수를 표시하고자 합니다.
![2017-09-13 4 02 23](https://user-images.githubusercontent.com/4696328/30364880-f5f2b01c-98a0-11e7-84f7-6ac3015f6d4e.png)
2. 주문서에서 미리준비를 누르면 미리 준비해야 하는 당일의 목록을 보여줍니다. 다음 표를 보고 미리 한번에 의류를 준비하고자 합니다. 따라서 아래와 그림과 같이 대여시간, 대여희망품목, 의류코드와 사이즈, 의류 상태가 필요합니다. 의류 사이즈는 다음과 같습니다. 

1) 남자 자켓 : 윗배 - 팔길이 - 가슴, 남자바지 : 허리 - 허벅지 - 바지길이
2) 여자 자켓 : 윗배 - 팔길이 - 가슴, 여자바지 : 허리 - 허벅지 - 바지길이, 여자 스커트 : 엉덩이 - 허리 - 기장 

![2017-09-13 4 02 11](https://user-images.githubusercontent.com/4696328/30365006-5f27c1c6-98a1-11e7-841b-bfdc3bac2318.png)

**Comment by 10001if** [comment]
미리준비의 주문서에는 반납예정일과 담당자가 나타나지 않을 것입니다.

**Comment by keedi** [comment]
진행합니다.

**Comment by keedi** [comment]
@10001if 
#1102 이슈와 제목이 유사해보이는데 어떤 관계와 어떤 차이가 있는지 확인 부탁드립니다. 필요하다면 #1067 과의 관계와 차이도 정리 부탁드립니다.

**Comment by 10001if** [comment]
저도 읽다보니 제가 쓰고도 좀 헷갈렸네요. 정리하자면 다음과 같습니다.

본 이슈와 #1067 과는 연속적이지만 개별 이슈입니다.
1067 이슈에서 요청드린 내용은 대여를 했던 대여자가 다시 방문하는 경우 지난 번에 대여 기록 중 자신이 다시 착용하고자 하는 의류를 입력하여 옷장지기에게 정보를 전달하는 이슈였습니다. 이를 위해 @aanoaa 작업해주셨고 현재는 아래 그림처럼 지난 대여기록을 확인해서 정보를 입력할 수 있는 란이 생성되었습니다.
![2017-11-17 12 40 16](https://user-images.githubusercontent.com/4696328/32928439-47d4dcda-cb95-11e7-929e-d58e424c0a27.png)

하지만 그 뒤에 입력한 정보를 옷장지기가 이해하고 대여를 진행하는 과정에서 정보 전달이 원활하지 않아 실제로 사용되기에는 부족한 점이 있었습니다. 그래서 1067 이슈의 하단에는 그 이후에 전달된 정보를 옷장지기가 어떻게 확인할 수 있고 확인된 내용을 어떻게 적용할 수 있을지에 대한 내용이었습니다. @keedi 이야기 주신 것처럼 피드백에 따른 개선 내용은 별도의 이슈를 열어달라고 하셔서 나머지 부분에 대한 내용을 다시 정리하였습니다.

**Comment by 10001if** [comment]
그래서 본 이슈는 #1067 을 진행하면서 발생한 피드백 이슈입니다.

**Comment by keedi** [comment]
@10001if 네, 개선 등 릴리스 이후 이슈는 별도로 열어서 작업하는 것이 편리합니다. 그래서 #1067의 피드백인 것은 이해했습니다. #1102와의 차이도 설명 부탁드립니다.

**Comment by 10001if** [comment]
그리고 #1102 이슈와는 일부 내용이 중복된 부분이 있습니다.

1102의 이슈는 몇가지의 프로세스를 묶어 개선하려고 한 내용입니다.

1. 입어본 의류를 다시 방문해서 대여하고 싶음
2. 입어본 의류를 온라인으로 쉽게 주문하고 싶음
3. 입어본 의류를 택배(퀵서비스)로 받고 싶음

이 3가지 부분을 개선하려고 한 것인데요.

본 이슈에서 다루는 내용과 `입어본 의류를 다시 방문해서 대여하고 싶음`과 유사한 부분이 있기는 하지만
본 이슈에서 다루는 내용은 이미 대여를 진행한 정상적인 절차를 진행한 대여자가 다시 방문했을 때 대여를 용이하게 하기 위한 것입니다.

1102에서 다루는 `입어본 의류를 다시 방문해서 대여하고 싶음`에서는 실제 결제를 마치지 않고 정말 입어만 보고 재 방문 하는 사람들에 관한 내용입니다. 

1102와 본 이슈 모두 크게 보면 재방문자에 대한 빠르고 신속한 대여 프로세스에 대한 내용입니다. 궁금한 점 있으시면 다시 남겨주세요.

**Comment by keedi** [comment]
@10001if 넵~ 기록해주신 부분 다시 복기해보고 필요한 부분있으면 문의드리겠습니다. 제목을 서로 구분이 될 수 있도록 잘 정리 해야할 것 같네요.

**Comment by 10001if** [comment]
네.. 제가 그냥 복사해서 붙여넣기를 해서 이런 결과가 발생한 듯 하네요.. ㅎㅎ 제목도 내용확인이 잘 될 수 있도록 신경써서 작성하겠습니다.

**Comment by pingpongworks** [comment]
성수기를 대비하기 위해, 
이전의 의류를 원하는 대여자의 경우, 
방문전에 미리 의류를 준비하고, 
입어보는 과정을 단축하자는 의견이 다시 나와서 
찾아보니까 이슈내용이 이미 있네요. 
진행상황이 궁금합니다.

**Comment by aanoaa** [comment]
이게 내부프로세스와도 관련이 있던 이슈라서 우선 원하는 방향을 잡아야 할 것 같습니다.
지난번에도 진행하다 말았던 이슈입니다.
해결을 위해서 제가 주말에 직접 찾아갈게요.(시간이 되시나요?)
그밖에 성수기를 대비한 이슈의 우선순위를 좀 생각해두시면 도움이 될 것 같아요.

**Comment by aanoaa** [comment]
제가 제대로 이해했다면 지난대여기록이 선택된 주문서에 대해 시간표에서 표시가 될 수
있도록 해서, 미리 의류를 준비할 수 있도록 하고 방문했을때에 정상적인 대여 프로세스를
거치지 않고 빠르게 의류를 대여하는 방법에 대한 것 입니다.

**Comment by pingpongworks** [comment]
>(시간이 되시나요?)

토요일에 출근하니까 시간이 됩니다!
오전에 신입 옷장지기 교육 끝나고 12시부터도 가능하고, 
오후 대여업무 끝난 후 6시 이후로도 가능합니다. 
온라인에 대해서도 전체적으로 정리가 필요할 거 같은데,  
우선순위를 좀 정리해 놓겠습니다~ @aanoaa

**Comment by pingpongworks** [comment]
>제가 제대로 이해했다면 지난대여기록이 선택된 주문서에 대해 시간표에서 표시가 될 수
>있도록 해서, 미리 의류를 준비할 수 있도록 하고 방문했을때에 정상적인 대여 프로세스를
>거치지 않고 빠르게 의류를 대여하는 방법에 대한 것 입니다.

맞습니다. 대여자가 예약시 이전 의류선택을 했더라도 
대여자가 방문을 하기 전까지는 옷장지기가 알 수 없기 때문에 
시간표에서 알아볼 수 있도록 표시가 되었으면 합니다. 
1차적으로 요청드리고 싶은 부분은 시간표 표시입니다. 

이후 정상적인 대여프로세스를 거치지 않고 빠르게 의류를 대여하는 방법에 대해서는
그동안 대여시스템 변동, 서울시쿠폰대여자 이용대장 작성 등 대여과정상 변화가 있었으므로
다시 한번 정리가 필요할 것 같습니다. @aanoaa

**Comment by aanoaa** [comment]
열린옷장의 정책상 예약이 가능하지 않습니다.
하지만 재대여이고 지난 대여기록중 원하는 의류가 있다면 daily 로 의류를 포장해두거나 준비해두는 기능입니다.
(정확하지 않습니다. 어떻게 할지 정해야 합니다.)

초기의 (전)한대표님의 아이디어입니다.

1.  대여자가 방문예약시에 지난대여이력을 선택한다.
2.  staff 서비스의 시간표에서 날짜별로 준비가능한 의류의 수를 남여별로 표시한다.
3.  staff 서비스의 시간표에서 준비가능한 주문서의 경우 구분가능하게 표시한다.
4.  staff 서비스의 주문서목록에서 미리 준비 가능한 주문서의 목록을 제공한다.
5.  대여자가 현장방문을 했을때 의류가 준비되어 있다면 치수측정이 아닌 탈의 혹은 그 이상의 단계로 넘어간다.

이렇게 준비가 된다면, 옷장지기는 매일 준비가 가능한 의류를 준비해두고 상태를 변경합니다.
대여자가 방문했을때에 준비된 의류를 바로 입어보거나 포장해서 대여함으로 대여시간을 단축합니다.

**Comment by aanoaa** [comment]
위에 상세하게 말씀해주셨지만 제 생각으로는 이게 꼭 시간표에 들어가야 하는 건 아니라고 생각합니다.
어디에 어떤 기능이 필요한지는 빼고 필요로하는 기능은 아래와 같습니다.

1.  날짜별로 의류준비가 가능한 주문서목록과 의류준비에 필요한 데이터를 제공
2.  옷장지기는 `1.` 을 참고해서 의류를 패키징하고 해당 주문서의 데이터를 손쉽게 변경
    (상태가 될지 뭐가 될지 모르겠습니다. 이 주문서는 패키징이 되었다는 것을 마킹해두는 것 입니다)
3.  대여자가 방문했을때에 패키징 되어 있다면 상태가 (의류준비/탈의/포장) 으로 변경

staff 서비스의 주문서 목록에서 의류를 준비하기 위한 데이터를 표시하기 적절하지 않을 것 같습니다.
주문서의 목록이 staff 가 아니라 monitor 의 의류준비와 비슷한 형태의 목록이 된다면 더 좋을 것 같습니다.

**Comment by aanoaa** [comment]
요구사항을 보면 지난 대여이력중에 선택한 주문서 뿐만 아니라, 방문해서 미리 입어본 경우에도 대응할 수 있어야 합니다.
대여자 입장에서는 미리 방문해서 입어본 후에 "다음에 왔을때 이 옷 또는 이 옷과 유사한 옷을 주세요."
라고 얘기하는 행위가 끝입니다.
이를 바탕으로 다음 방문때에 의류가 준비되어 있어야 합니다.

지난 대여이력을 선택하는 것은 이번 방문예약때에 대여-반납이 이루어진 주문서중에 하나를 선택합니다.
옷장지기가 입력하는 과정은 없습니다. 이는 포장/결제할때에 이미 입력이 된 데이터 입니다.

하지만 입어보는 경우라면 어떤 주문서에 입어본 의류를 입력하는 과정이 필요하고 입력된 이후에는
대여자가 대여이력을 선택한 것과 같은 과정이 시스템에서 자동으로 이루어져야 합니다.

**Comment by aanoaa** [comment]
입어보고 가는 대여자들의 문제는 더 고민해봐야합니다.

의류를 입어보기 위해서

1.  예약없이 방문
2.  예약일 보다 먼저 방문
    -   한번 예약
    -   두번 예약(입어보기위한 예약 + 대여일 예약)
3.  예약일에 방문

하는 경우가 있을 수 있습니다.
**예약없이 방문했을때에는 어떻게 처리하고 계신가요?**

의류를 입어본 경우에 보통 다시 방문하지 않고 택배를 통해서 받기를 원할 것 같습니다.
그렇게 된다면 **의류를 언제 패키징 하는게 좋을까요?**

**Comment by pingpongworks** [comment]
@aanoaa 이 이슈는 그 동안 논의한 내용에 연결해서 생각을 발전시키기가 좀 어려운 듯 합니다. 현 시점에서  옷장지기들이 대여자를 실제 어떻게 응대하고 있는지를 바탕으로 좀더 심플하게 프로그램과 관련된 개선점을 다시 정리해보는 것이 효율적이지 않을까 합니다.  대여현장에서 원하는 프로세스가 정리되면 다시 요청드리겠습니다.
